transport/http/header.go

```diff
@@ -44,12 +44,30 @@ func (hm headerMapper) ToHTTPHeaders(from transport.Headers, to http.Header) htt
 	if to == nil {
 		to = make(http.Header, from.Len())
 	}
-	for k, v := range from.Items() {
-		to.Add(hm.Prefix+k, v)
+	// We use the original header-key, if the key is written in a MIME canoicalized form we save an allocation.
+	for k, v := range from.OriginalItems() {
```

Comment:
Could you pull this out into a separate PR? Most likely with the appopriate benchmarks?

(or, have a totally separate PR for benchmarks too, then follow with this change?)

Seems important/large enough to call it out separately.